### PR TITLE
[PyTorch] Enable generic QK norm support (+ RMSNorm/LayerNorm)

### PR DESCRIPTION
**Pull request by negvet**
# Description

For the training stabilization purposes, QK tensors might be normalized.

- Enables generic QK norm support.
- Enables `RMSNorm`/`LayerNorm` as a normalization types (in addition to `L2Normalization`).
- Enables qk normalization before and after `RoPE` (following both Qwen and Llama approaches).

Extention of https://github.com/NVIDIA/TransformerEngine/pull/1864
Fixes [#1958](https://github.com/NVIDIA/TransformerEngine/issues/1958)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/pytorch/attention/multi_head_attention.py [hunk 1]
```diff
@@ -175,14 +175,17 @@ class MultiheadAttention(torch.nn.Module):
                     parameter for query-key-value. This enables optimizations such as QKV
                     fusion without concatentations/splits and also enables the argument
                     `fuse_wgrad_accumulation`.
-    use_qk_norm: bool, default = 'False'
-                    if set to `True`, L2 normalization is applied to query and key tensors
-                    after RoPE (if applicable) but before attention computation.
-                    This follows the Llama4 approach for QK normalization to improve
+    qk_norm_type: Optional[str], default = None
```
Comment (Marks101) [inline transformer_engine/pytorch/attention/multi_head_attention.py:178]: In other parts of the code, the key for RMS normalization is 'RMSNorm', here it is now 'rms'. Maybe it is worth to consider adopting this.

Additionally, I was wondering if it makes sense to support LayerNorm was well. Simply to have everything feature complete.

Comment (negvet) [reply transformer_engine/pytorch/attention/multi_head_attention.py:178]: Good points, fixed.

**Comment by negvet** [comment]
> I have to admit that there is one more thing that I just noticed. The Qwen3 models apply QK normalization before RoPE, see [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen3/modular_qwen3.py#L78-L83), this is in contrast to this implementation which is based on LLama4. I was not aware that there are two different formulations for this. Sorry for that.

This flexibility might be worth it; I supported it.
It brings some complexity and additional argument, but it is manageable, I guess.
Let's give it a review.
